### DISCUSSION TITLE
Site Permissions for Focus

### DISCUSSION THREAD

**Discussion by amedyne** [mozilla-mobile/focus-android#5830] [General]
This discussion is to hash out plans and details for providing site permission settings for end users on Focus.
Tagging @channingcarter and @jeffreygee .

This is a follow-up to the following GH ticket: https://github.com/mozilla-mobile/focus-android/issues/5743

- Where do site permissions in the browser view live? 

- Will site permissions persist for a browser that is supposed to not save your browsing history? (i.e. to enable/disable a permission when you visit it again, we store the pref of the site)

- Should we review what the defaults for each permission should be to meet Focus' needs?

**Comment by jonalmeida** [comment]
@Amejia481 and I spoke offline about this too: it wasn't clear to me earlier in the meeting, but is the ask to only expose the autoplay permissions? Should we consider permissions for other settings too? It seems reasonable to want to allow location for a site temporarily while you are shopping or using Maps.

**Reply by channingcarter** [reply]
Yes we want to expose autoplay site permissions similar to the way Fenix exposes them.  But we do want to create a UI for other settings as well. Jeff and I will talk about what those should be. But for now let's start with autoplay since we can duplicate what Fenix has. @jonalmeida @jeffreygee @amedyne

**Reply by jonalmeida** [reply]
Great, thanks for clarifying! 

Sounds like we need to add the same UI in these screenshots for **only autoplay but with the intention of supporting the others too**. I think it should be similar amounts of eng effort for just autoplay as it would be for all since we would just make the other settings invisible for now.

<img width="1384" alt="out" src="https://user-images.githubusercontent.com/1370580/141004458-838eccd7-c194-4463-bd79-33e59f87899f.png">

**Reply by jonalmeida** [reply]
@amedyne I'm not sure who is going to work on this, but to avoid us doing duplicate work in Focus for site permissions, we should check to see if there are things in Fenix's site permissions that we should upstream to AC so that Focus gets it for free as well.

**Comment by iorgamgabriel** [comment]
Site Permissions were implemented on Focus . I will arhive this conversation.
